### PR TITLE
End of game

### PR DESCRIPTION
**Pull request by zo-el**

### CODE REVIEW

**Review by willemolding — APPROVED** [review]
Looks great. I only reviewed the CellMatrix stuff so maybe someone else should review the UI code as well

File: ui-src/src/CellMatrix.ts [hunk 1]
```diff
@@ -116,6 +117,20 @@ export default class CellMatrix {
     return (this.getValue(pos) & 0b00100000) > 0;
   }
 
+  public isCompleted(): boolean {
+    for (let y = 0; y < this.size.y; y++) {
+      for (let x = 0; x < this.size.x; x++) {
+        const pos = {x,y}
+         // console.log("checking: ",pos,this.isRevealed(pos),this.isFlagged(pos))
+        if (!this.isRevealed(pos) && !this.isFlagged(pos)) {
+         // console.log("not revealed yet: ",pos,this.isRevealed(pos),this.isFlagged(pos))
```
Comment (maackle) [inline ui-src/src/CellMatrix.ts]: Remove comments first?

File: ui-src/src/store.tsx [hunk 2]
```diff
@@ -41,9 +41,12 @@ function reduceGame (state: StoreGameState, action: ReduxAction) {
       break;
     }
   }
+    const gameOver:boolean = matrix.isCompleted();
+    // console.log("reduceGame",{...state,gameOver,matrix});
```
Comment (maackle) [inline ui-src/src/store.tsx]: remove?

File: ui-src/src/store.tsx [hunk 3]
```diff
@@ -53,25 +56,26 @@ export function reducer (oldState: StoreState = defaultState, action: ReduxActio
     currentGame: reduceGame(oldState.currentGame, action),
     myActions: oldState.myActions + 1,
   }
-
+  // console.log("state",state)
```
Comment (maackle) [inline ui-src/src/store.tsx]: remove?

**Review by maackle — APPROVED** [review]
Looks good! Also it would be nice to remove some commented out code to keep it clean

**Pull request merged**